src/Tes/Repository/PostgreSqlCachingRepository.cs

```diff
@@ -134,8 +134,8 @@ protected Task<T> AddUpdateOrRemoveItemInDbAsync(T item, WriteAction action, Can
                 }
             }
 
-            _itemsToWrite.Enqueue((item, action, source));
-            return result;
+            await _itemsToWrite.Writer.WriteAsync((item, action, source));
+            return item;
```

Comment:
Do we need to add `await result;` above this line?

Reply:
I would revert lines 120 and 138 and remove the await on 137. The purpose of the implementation of this method is to return a Task<T> representing the writing of the task to the DB.

If you insist on having this method be async (overall the whole system will be faster and have smaller code if it is not) then you should replace line 138 with `var returnValue = await result;\n return returnValue;`